msamp/common/include/common.h

```diff
@@ -68,4 +106,24 @@ template <> __host__ __device__ bf16 cast_dtype(const float value) { return __fl
 
 template <> __host__ __device__ float cast_dtype(const fp16 value) { return __half2float(value); }
 
-template <> __host__ __device__ float cast_dtype(const bf16 value) { return __bfloat162float(value); }
\ No newline at end of file
+template <> __host__ __device__ float cast_dtype(const bf16 value) { return __bfloat162float(value); }
+
+template <typename T>
+struct is_fp8 : std::false_type {};
+
+template <>
+struct is_fp8<fp8e4m3> : std::true_type {};
+
+template <>
+struct is_fp8<fp8e5m2> : std::true_type {};
+
+template <typename T>
+struct is_fp16 : std::false_type {};
+
+template <>
+struct is_fp16<fp16> : std::true_type {};
+
+template <>
+struct is_fp16<bf16> : std::true_type {};
```

Comment:
It is ambiguous that bf16 is fp16. I prefer to rename it to `is_fp16_or_bf16`

Reply:
Good point. Maybe `is_half` is better.